src/main/scala/com/github/symcal/Expr.scala

```diff
@@ -93,15 +94,14 @@ case class Var(name: Symbol) extends Expr {
 case class IntPow(x: Expr, d: Int) extends Expr {
   override def toInt: Int = Math.pow(x.toInt, d).toInt
 
-  override def diff(z: Var): Expr = d match {
-    case 0 => 0
+  override def diff(z: Var): Expr = (d match {
+    case 0 => Const(0)
```

Comment:
Because it doesn't know that it should insert `intToConst`. The result of the entire `match` expression previously had type `Any`! This was definitely a bug. We didn't know this until I wanted to compute `.simplify` of diff. Another possibility to fix this would be to write
```scala
 case 0 => 0 : Expr
```

would you prefer this?